src/command.rs

```diff
@@ -271,226 +305,351 @@ pub fn parse(input: &str) -> Option<Vec<Command>> {
-                    .ok()
-                    .or_else(|| {
-                        parse_duration::parse(&duration_raw) // accept fancy duration
-                            .ok()
-                            .and_then(|dur| dur.as_millis().try_into().ok())
-                    })
-                    .and_then(|unsigned_millis| {
-                        match first_char {
-                            // handle i32::MAX < unsigned_millis < u32::MAX gracefully
-                            Some('+') => {
-                                i32::try_from(unsigned_millis)
-                                    .ok()
-                                    .map(|unsigned_millis_i32| {
-                                        SeekDirection::Relative(unsigned_millis_i32)
-                                    })
-                            }
-                            Some('-') => {
-                                i32::try_from(unsigned_millis)
-                                    .ok()
-                                    .map(|unsigned_millis_i32| {
-                                        SeekDirection::Relative(-unsigned_millis_i32)
-                                    })
-                            }
-                            _ => Some(SeekDirection::Absolute(unsigned_millis)),
-                        }
-                        .map(|direction| Command::Seek(direction))
-                    })
+                let unsigned_millis = match duration_raw.parse() {
+                    // accept raw milliseconds
+                    Ok(millis) => millis,
+                    Err(_) => parse_duration::parse(&duration_raw) // accept fancy duration
+                        .map_err(|err| ArgParseError {
+                            arg: duration_raw.clone(),
+                            err: err.to_string(),
+                        })
+                        .and_then(|dur| {
+                            dur.as_millis().try_into().map_err(|_| ArgParseError {
+                                arg: duration_raw.clone(),
+                                err: "Duration value too large".into(),
+                            })
+                        })?,
+                };
+                let seek_direction = match first_char {
+                    // handle i32::MAX < unsigned_millis < u32::MAX gracefully
+                    Some('+') => {
+                        i32::try_from(unsigned_millis).map(|millis| SeekDirection::Relative(millis))
+                    }
+                    Some('-') => i32::try_from(unsigned_millis)
+                        .map(|millis| SeekDirection::Relative(-millis)),
+                    _ => Ok(SeekDirection::Absolute(unsigned_millis)),
+                }
+                .map_err(|_| ArgParseError {
+                    arg: duration_raw,
+                    err: "Duration value too large".into(),
+                })?;
+                Command::Seek(seek_direction)
+            }
```

Comment:
@hrkfdn I would like to know your opinion on keeping backward compatibility for `seek [+|-]<milliseconds>`.  It caused me a lot of headache trying to write this code cleanly, but I still ended up with this mess.

If you think keeping backward compatibility here is not so important, I'm happy to go ahead and remove it.

Reply:
Hmm, I don't think it's so bad :smile: 

Ideally we should set the default unit to `ms` or fail when values without units are supplied, otherwise existing custom bindings will confuse users a lot, as they may suddenly seek forward by a couple of hundred seconds. So maybe we should keep it backward compatible for now? WDYT?

Reply:
The reason I suggested to make `second` the default unit is actually quite simple... `parse_duration` crate doesn`t provide any options to configure its default unit, which means we either have to use its default (second), or do the dance we've got here to try parsing to `ms` first.

If you think the code isn't too terrible I'm happy to keep it as is.